core/contextmenu_registry.ts

```diff
@@ -71,56 +72,60 @@ export class ContextMenuRegistry {
   }
 
   /**
-   * Gets the valid context menu options for the given scope type (e.g. block or
-   * workspace) and scope. Blocks are only shown if the preconditionFn shows
+   * Gets the valid context menu options for the given scope.
+   * Options are only included if the preconditionFn shows
    * they should not be hidden.
    *
-   * @param scopeType Type of scope where menu should be shown (e.g. on a block
-   *     or on a workspace)
    * @param scope Current scope of context menu (i.e., the exact workspace or
-   *     block being clicked on)
+   *     block being clicked on).
+   * @param menuOpenEvent Event that caused the menu to open.
    * @returns the list of ContextMenuOptions
    */
   getContextMenuOptions(
-    scopeType: ScopeType,
     scope: Scope,
     menuOpenEvent: Event,
   ): ContextMenuOption[] {
     const menuOptions: ContextMenuOption[] = [];
     for (const item of this.registeredItems.values()) {
-      if (scopeType === item.scopeType) {
-        let menuOption:
-          | ContextMenuRegistry.CoreContextMenuOption
-          | ContextMenuRegistry.SeparatorContextMenuOption
-          | ContextMenuRegistry.ActionContextMenuOption;
+      if (item.scopeType) {
+        // If the scopeType is present, check to make sure
+        // that the option is compatible with the current scope
+        if (item.scopeType === ScopeType.BLOCK && !scope.block) continue;
+        if (item.scopeType === ScopeType.COMMENT && !scope.comment) continue;
+        if (item.scopeType === ScopeType.WORKSPACE && !scope.workspace)
+          continue;
+      }
```

Comment:
github is not showing a good diff, the only difference from this point on is that the if statement ends here and everything else is unindented correspondingly

Reply:
Much better with Hide Whitespace on: 
![image](https://github.com/user-attachments/assets/31642e6d-7840-4e12-aa7f-26141c7c9e71)